pkg/client/william.go

```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+)
+
+// William Interface
+type William interface {
```

Comment:
Don't define the interface yourself - in Go we usually push interface declaration to the _user_ of the code, not the _declarer_. We should be exposing a concrete type here and, if whomever uses it wants an interface, they can declare it as they see fit.